src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -196,16 +223,8 @@ public PaymentHandleResponse executePayment(@Nonnull String paypalPlusPaymentId,
                         return CompletableFuture.completedFuture(PaymentHandleResponse.of404NotFound(
                                 format("Can't find cart with interfaceId==[%s]", paypalPlusPaymentId)));
                     } else {
-                        Cart cart = cartOpt.get();
-                        PaymentHandleResponse paymentHandleResponse = patchAddress(cart, paypalPlusPaymentId);
-                        boolean isSuccessful = HttpStatus.valueOf(paymentHandleResponse.getStatusCode()).is2xxSuccessful();
-                        if (isSuccessful) {
-                            // execute payment only when patching was successful
-                            return updatePayerIdInCtpPayment(paypalPlusPaymentId, paypalPlusPayerId)
-                                    .thenCompose(ctpPayment -> executePaymentAndCreateTxn(paypalPlusPaymentId, paypalPlusPayerId, ctpPayment));
-                        } else {
-                            return CompletableFuture.completedFuture(paymentHandleResponse);
-                        }
+                        return updatePayerIdInCtpPayment(paypalPlusPaymentId, paypalPlusPayerId)
```

Comment:
the payment in `updatePayerIdInCtpPayment()` not found by id, the value in the stage will be null, but chained `executePaymentAndCreateTxn()` accepts _ctpPayment_ only as _@Nonnull_ => NPE prone solution

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/51/commits/9673a67ac2ff6b29b3ee4e35a124b8f051f1ce0f

Reply:
You misunderstood me: i complain that next chained function `executePaymentAndCreateTxn()` will fail if payment in `updatePayerIdInCtpPayment()` not found

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/51/commits/74e58720b5606484cc91819dfa6dc5f55174221d